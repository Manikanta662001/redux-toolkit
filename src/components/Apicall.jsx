import React from "react";
import { useDispatch, useSelector } from "react-redux";
import { fetchUsers } from "../store/slices/apiSlice";

function Apicall() {
  const dispatch = useDispatch();
  const state = useSelector((state) => console.log(state));
  const users = useSelector((state) => state.apiReducer.users);
  return (
    <div>
      <button onClick={() => dispatch(fetchUsers())}>Api Call</button>
      <ul>
        {users?.map((user) => {
          return <li key={user.id}>{user.name}</li>;
        })}
      </ul>
    </div>
  );
}

export default Apicall;
