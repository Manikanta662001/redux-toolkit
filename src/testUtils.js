import { combineReducers, configureStore } from "@reduxjs/toolkit";
import { fetchUsers } from "./store/thunks/fetchUsersThunk";
import counterReducer from "./store/slices/counterSlice";
import apiReducer from "./store/slices/apiSlice";
import { render } from "@testing-library/react";
import storage from "redux-persist/lib/storage";
import { persistReducer } from "redux-persist";
import { Provider } from "react-redux";

export const renderWithProvider = (
  ui,
  preloadedState = {},
  customReducers = {}
) => {
  const rootReducer = combineReducers({
    apiReducer,
    counterReducer,
    ...customReducers,
  });
  const rootResetReducer = (state, action) => {
    if (action.type === "RESET") {
      return {
        ...state,
        counterReducer: undefined,
        apiReducer: undefined, // Reset this part of the state
      };
    }
    return rootReducer(state, action);
  };
  const persistConfig = {
    key: "root",
    storage,
    whitelist: ["counterReducer"],
  };

  const persistedReducer = persistReducer(persistConfig, rootResetReducer);

  const store = configureStore({
    reducer: persistedReducer,
    extraReducers: (builder) => {
      builder.addCase(fetchUsers.pending, (state) => {
        return { ...state, isLoading: true };
      });
      builder.addCase(fetchUsers.fulfilled, (state, action) => {
        return { ...state, isLoading: false, users: action.payload };
      });
      builder.addCase(fetchUsers.rejected, (state, action) => {
        return { ...state, isLoading: false, error: action.payload };
      });
    },
    middleware: (getDefaultMiddleware) =>
      getDefaultMiddleware({
        serializableCheck: {
          ignoredActions: [
            "persist/PERSIST",
            "persist/REHYDRATE",
            "persist/PAUSE",
            "persist/PURGE",
            "persist/FLUSH",
            "persist/REGISTER",
          ],
        },
      }),
    preloadedState,
    devTools: true,
  });
  if (ui === undefined) return store;
  return { ...render(<Provider store={store}>{ui}</Provider>), store };
};
