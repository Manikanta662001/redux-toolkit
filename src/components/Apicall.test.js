import { renderWithProvider } from "../testUtils";
import Apicall from "./Apicall";

describe("Counter Component", () => {
  const initialState = {
    counterReducer: { count: 0 },
    apiReducer: {
      users: [],
      isLoading: false,
      error: "",
    },
  };

  test("renders correctly and Match the Snapshot", () => {
    const { asFragment } = renderWithProvider(<Apicall />, initialState);
    expect(asFragment(<Apicall />)).toMatchSnapshot();
  });
});
