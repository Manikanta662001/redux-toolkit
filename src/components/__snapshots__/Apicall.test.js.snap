// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter Component renders correctly and Match the Snapshot 1`] = `
<DocumentFragment>
  <div>
    <button>
      Api Call
    </button>
    <ul />
  </div>
</DocumentFragment>
`;
